src/matrix.c

```diff
@@ -141,6 +146,12 @@ int ir_matrix_identity_lua(lua_State *L) {
 int ir_matrix_index_lua(lua_State *L) {
 	const char *key;
 
+    // I think this should not be done like this and should be implemented
+    // through the kernel internals, to avoid cascading strcmp in favor of
+    // hash tables.
+    //
+    // I'm gonna leave it like this for now tho ~FabricatorZayac
```

Comment:
Feel free to optimize this. As long as you can eventually index the individual components of a matrix, I'm all for it.

Example: matrix[c][r]

P.S. Are matrices supposed to be column-major or row-major?